feed foundation wallet by default, not faucet

joshuef opened the pull request:
## Description

reviewpad:summary

davidrusu commented on sn_node/src/node/bootstrap/join.rs:
```diff
@@ -358,22 +358,23 @@ impl<'a> Joiner<'a> {
-                .max_by_key(|(age, count)| (*count, -(**age as i32)))?;
-            if self.node.age() != *voted_expected_age {
-                return Some(*voted_expected_age);
+                .find(|(_age, count)| **count >= threshold as u8)
+                .map(|(age, _count)| *age)?;
```
Don't you want to prefer the smaller age if you have multiple ages that have passed the threshold? i.e.
```suggestion
                .filter(|(_age, count)| **count >= threshold as u8)
                // In case two ages got the same votes, we prefer the lower age
                .max_by_key(|(age, count)| (*count, -(**age as i32)))?;
```

maqi replied:
the threshold is 2/3 of the elders.len(), and the votes shall all from the same elders, which means there shall only be one `age` got more than 2/3 elders' votes.
so no need to have the `max_by_key` anymore

davidrusu replied:
we talked through this, the point is that we always track the most-recent votes from each elder, so it's impossible to get multiple super-majorities here.

jacderida commented on .github/workflows/sn_pr_tests.yml:
```diff
@@ -90,10 +90,17 @@ jobs:
               - 'sn_node/src/**'
+              - 'sn_node/benches/**'
+              - 'sn_node/examples/**'
+              - 'testnet/**'
```
Yeah, I would be for that. I think this was originally used for another purpose than controlling just Clippy. Maybe for running particular tests or something. That doesn't seem to apply any more, so let's just get rid of it.

maqi commented:
bors r+

iancoleman committed:
feat(cli): eip2333 helpers for accounts

joshuef committed:
feat(transfers): do not genereate wallet by default

joshuef committed:
feat(cli): generate a mnemonic as wallet basis if no wallet found

joshuef committed:
feat(faucet): add feat for gifting-from-genesis

So we no longer gift the entire genesis amount to the faucet
(which we can fund in other means).

but we retain a feat for this for testing etc

joshuef committed:
ci: update got 'gifting-from-genesis' faucet feat

joshuef committed:
fix(client): ensure we have a wallet or generate one via mnemonic

joshuef committed:
fix(client): calm down broadcast error logs if we've no listeners

joshuef committed:
fix(client): move acct_packet mnemonic into client layer

Allows for reuse in faucet works and proper ci usage.

Makes the account packet tooling more client layer there.

joshuef committed:
chore(transfers): reduce error size

joshuef committed:
fix(client): set uploader to use mnemonic wallet loader

joshuef committed:
feat(cli): readd wallet helper address for dist feat

joshuef committed:
fix(faucet): fix distribution 'from' wallet loading

joshuef committed:
chore: devSkim ignore foundation pub temp key

joshuef committed:
fix: update calls to HotWallet::load

Use account packet helpers to generate wallet reliably now  errors out if
no wallet is found

joshuef committed:
fix: transfer tests for HotWallet creation

joshuef committed:
fix: more test and cli fixes

joshuef committed:
fix: create faucet via account load or generation

joshuef committed:
fix(cli): acct_packet tests updated

maqi committed:
chore(faucet): log initilization failure and upload faucet log

maqi committed:
chore(faucet): log initilization failure and upload faucet log

joshuef committed:
fix(faucet): ensure faucet is funded in main fn

joshuef committed:
ci: add gifting-from-genesis feat for workflows

joshuef committed:
chore: small cleanup of dead code

joshuef committed:
chore(cli): update mnemonic wallet seed phrase wording

joshuef committed:
fix(faucet): rate limit before getting wallet

joshuef committed:
fix(faucet): cleanup unused vars
